### PR TITLE
LUCENE-10377: Replace 'sortPos' with 'enableSkipping' in SortField.getComparator()

### PR DESCRIPTION
**Pull request by romseygeek**

### CODE REVIEW

**Comment by romseygeek** [comment]
This might make it possible to remove the `getPointsValues()` override from NumericLeafComparator as well as a follow-up, cc @mayya-sharipova

File: lucene/core/src/java/org/apache/lucene/search/comparators/NumericComparator.java [hunk 1]
```diff
@@ -153,26 +152,6 @@ protected NumericDocValues getNumericDocValues(LeafReaderContext context, String
...
-     * points and doc values record the same information. Return {@code null} even if no points
-     * implementation is available, in this case sort optimization with points will be disabled.
-     *
-     * @param context – reader context
-     * @param field - field name
-     * @return point values for the field in this segment if they are available or {@code null} if
-     *     sort optimization with points should be disabled.
-     * @throws IOException If there is a low-level I/O error
-     */
-    protected PointValues getPointValues(LeafReaderContext context, String field)
```
Comment (mayya-sharipova) [inline lucene/core/src/java/org/apache/lucene/search/comparators/NumericComparator.java:171]: I can't think of any case where we have a custom representation of docValues and also provide a custom representation of pointValues. So this change is LGTM!

**Review by mayya-sharipova — APPROVED** [review]
@romseygeek  Thanks Alan, this API change LGTM.  Smart change, and I also liked how it simplified code in some places.

**Comment by mayya-sharipova** [comment]
> This might make it possible to remove the getPointsValues() override from NumericLeafComparator as well as a follow-up, cc @mayya-sharipova

@romseygeek I think you have already done this removal in this PR, haven't you?

**Pull request merged**